src/middlewared/middlewared/plugins/alert.py

```diff
@@ -377,6 +389,21 @@ def restore(self, id):
         except UnavailableException:
             raise CallError("This alert checker is unavailable", CallError.EALERTCHECKERUNAVAILABLE)
 
+    @private
+    async def block_source(self, source_name):
+        if source_name not in ALERT_SOURCES:
+            raise CallError("Invalid alert source")
+
+        lock = str(uuid.uuid4())
+        self.blocked_sources[source_name].add(lock)
```

Comment:
A thought that just crossed my mind, what if for whatever reason the client simply dies before it can unlock?
Should we have some sort of timer to autolock after a huge amount of time has passed? e.g. 5 or 10 min?

Reply:
Pool creation take take half an hour on one of our test systems in the lab, that's why I've set default timeout to 3600.